dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js

```diff
@@ -5,41 +5,47 @@ const { findOneDeal } = require('./get-deal.controller');
 const addDealComment = async (_id, commentType, comment) => {
   const collection = await db.getCollection('deals');
 
-  const findAndUpdateResponse = await collection.findOneAndUpdate(
-    { _id: ObjectId(_id) },
-    {
-      $push: {
-        [commentType]: {
-          $each: [{
-            ...comment,
-            timestamp: Date.now(),
-          }],
-          $position: 0,
+  if (ObjectId.isValid(_id)) {
```

Comment:
Do we need this condition here? It's done in the POST function below

Reply:
Noticed this in all the functions - I imagine it's fail safe with some of the functions that will be called without the "crud endpoint"? Happy to approve.

Reply:
exactly. it should be handled as soon as the endpoint is called. but sometimes we import those functions in other files and it skips that validation from "crud endpoint". So yes, it's a fail-safe :)

Reply:
Brilliant 😄